stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/textinput/MessageInputFieldView.kt

```diff
@@ -0,0 +1,259 @@
+package io.getstream.chat.android.ui.textinput
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.FrameLayout
+import androidx.annotation.ColorInt
+import androidx.annotation.Px
+import androidx.core.view.isVisible
+import androidx.core.widget.doAfterTextChanged
+import com.getstream.sdk.chat.model.AttachmentMetaData
+import com.getstream.sdk.chat.utils.StorageHelper
+import io.getstream.chat.android.client.models.Command
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.client.models.name
+import io.getstream.chat.android.ui.R
+import io.getstream.chat.android.ui.attachments.selected.SelectedFileAttachmentAdapter
+import io.getstream.chat.android.ui.attachments.selected.SelectedMediaAttachmentAdapter
+import io.getstream.chat.android.ui.databinding.StreamUiMessageInputFieldBinding
+import io.getstream.chat.android.ui.utils.extensions.EMPTY
+import io.getstream.chat.android.ui.utils.extensions.leftDrawable
+import io.getstream.chat.android.ui.utils.extensions.setTextSizePx
+import java.io.File
+import kotlin.properties.Delegates
+
+public class MessageInputFieldView : FrameLayout {
+    public val binding: StreamUiMessageInputFieldBinding =
+        StreamUiMessageInputFieldBinding.inflate(LayoutInflater.from(context), this, true)
+
+    private val attachmentModeHint: String = context.getString(R.string.stream_ui_message_input_field_attachment_hint)
+    private val normalModeHint = context.getString(R.string.stream_ui_message_input_field_message_hint)
+    private val selectedFileAttachmentAdapter: SelectedFileAttachmentAdapter = SelectedFileAttachmentAdapter()
+    private val selectedMediaAttachmentAdapter: SelectedMediaAttachmentAdapter = SelectedMediaAttachmentAdapter()
+    private val storageHelper = StorageHelper()
+
+    private var selectedAttachments: List<AttachmentMetaData> = emptyList()
+    private var contentChangeListener: ContentChangeListener? = null
+
+    public var mode: Mode by Delegates.observable(Mode.MessageMode) { _, _, newMode ->
+        when (newMode) {
+            is Mode.FileAttachmentMode -> switchToFileAttachmentMode(newMode)
+            is Mode.MediaAttachmentMode -> switchToMediaAttachmentMode(newMode)
+            is Mode.MessageMode -> switchToNormalMode()
+            is Mode.EditMessageMode -> switchToEditMode(newMode)
+            is Mode.CommandMode -> switchToCommandMode(newMode)
+        }
+        contentChangeListener?.onModeChanged(newMode)
+    }
+
+    public var messageText: String
```

Comment:
Just a thought, non-actionable for this PR: At some point we should really review the patterns we use for our public API in Views. Sometimes we expose setter functions, sometimes properties, etc. It's hardly consistent.